convokit/model/corpus.py

```diff
@@ -601,38 +614,60 @@ def filter_conversations_by(self, selector: Callable[[Conversation], bool]):
         }
         self.update_speakers_data()
         self.reinitialize_index()
+
+        # clear all storage entries corresponding to filtered-out components
+        meta_ids = []
+        for utt in self.iter_utterances():
+            meta_ids.append(utt.meta.storage_key)
+        for convo in self.iter_conversations():
+            meta_ids.append(convo.meta.storage_key)
+        for speaker in self.iter_speakers():
+            meta_ids.append(speaker.meta.storage_key)
+        self.storage.purge_obsolete_entries(
+            self.get_utterance_ids, self.get_conversation_ids(), self.get_speaker_ids(), meta_ids
+        )
+
         return self
 
-    def filter_utterances_by(self, selector: Callable[[Utterance], bool]):
+    @staticmethod
+    def filter_corpus_utterances(source_corpus: "Corpus", selector: Callable[[Utterance], bool]):
```

Comment:
nit: `Corpus.filter_corpus_utterances` -> `Corpus.filter_utterances` -- succinct + gets the point across just as well